### PR TITLE
add string comparison support

### PR DESCRIPTION
**Pull request by ozanh**
I'd like to add support for `< > <= >=` for strings. Tests and docs are added. Although bytecode contains only `> >=` operators `Int` object has `< <=` operator support so I added for consistency.

### CODE REVIEW

**Comment by d5** [comment]
can you merge/rebase your branch to the latest master?

**Comment by ozanh** [comment]
Is merge ok now? I was not using github often but I have been learning :smile: even at 36 yo :cry:

File: objects.go [hunk 1]
```diff
@@ -1342,6 +1342,50 @@ func (o *String) BinaryOp(op token.Token, rhs Object) (Object, error) {
 			}
 			return &String{Value: o.Value + rhsStr}, nil
 		}
+	case token.Less:
+		switch rhs := rhs.(type) {
+		case *String:
+			switch op {
```
Comment (d5) [inline objects.go:1348]: why do you check `op` once again?

Comment (ozanh) [reply objects.go]: my bad mistake sorry about that

**Comment by d5** [comment]
looks good. @geseq please review and feel free to merge it

**Pull request merged**